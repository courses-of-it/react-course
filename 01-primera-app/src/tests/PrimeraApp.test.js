import "@testing-library/jest-dom";

import React from "react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe("Pruebas en <PrimeraApp/>", () => {
  /*test('debe de mostrar el mensaje "hola soy goku"', () => {
	const saludo = 'hola soy goku'

	const { getByText } = render( <PrimeraApp saludo={saludo}/> )

	expect(getByText(saludo)).toBeInTheDocument()

    })*/
  test("debe de mopstrar <PrimeraApp /> correctamente", () => {
    const saludo = "hola soy goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar el subitulo mandado por props", () => {
    const saludo = "hola soy goku";
    const subtitulo = "subitulo";
    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
