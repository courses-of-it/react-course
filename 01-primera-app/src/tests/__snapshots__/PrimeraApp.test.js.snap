// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mopstrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hola soy goku
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
